### ISSUE TITLE
Non-integer membership prices

### ISSUE THREAD

**Issue by ssbarsky** [jordanbrown0/reg3#441]
Boskone has at least one class with a non-integer value. Virtual = $42.60. There are reason why this was useful for Treasury, ecomm searches, etc. Class configuration screen would not allow me to enter $24.60 as a value.

**Comment by jordanbrown0** [comment]
It's deliberately in even dollars, because it kept everything simpler and I thought that sane people wouldn't want to be playing with pennies.

If it's critical I can look into allowing dollars and cents.  (Or, more generally, whatevers and cents.  I don't think there are any common currencies that work with tenths or mils.)

**Comment by ssbarsky** [comment]
It's a Virtual membership so no one is playing with non-virtual pennies. :D
And for the very few people that might have showed up at-con and wanted to
upgrade from Virtual to Full Attending, we rounded up and gave them the
"bonus $.41". (There may have been one at B59) :D

It's not mission critical, I can find a way to fudge the $24.60, but it
would be nice to not have to do that.

Sharon Sbarsky
NESFA Vice President
www.nesfa.org
Boskone 60, February 18-20, 2022
Westin Boston Seaport District
www.boskone.org

**Comment by ssbarsky** [comment]
I'm able to have $24.60 in the class amount. (I think I edited the amount in csv before importing classes.) But when I tried to create and upgrade from "master import" to "virtual", there was an error:

![image](https://user-images.githubusercontent.com/75351842/211328513-67210616-433c-41b9-a5b4-fecc3827f7f8.png)

**Comment by jordanbrown0** [comment]
Q:  in your import, did you say "number" for the converter?  (It's the item in a field map entry.)  If you didn't say "number" then all kinds of entertaining things will happen, notably on upgrade.  "24.60" plus ten is not 34.60; it's 24.6010.  Ask me how I know this :-)

Hmm.  There's definitely a mismatch there; the importer allows "." but the input does not.  They should be the same, which really means that input should have an option for allowing decimals.

**Comment by ssbarsky** [comment]
I left the fields blank for the import.

**Comment by jordanbrown0** [comment]
Yeah, so they are probably strings.  Nothing bad will happen (other than this error) until you try to do arithmetic on them (probably via upgrade), and then it will be silly.

**Comment by jordanbrown0** [comment]
If you want a bunch of data repaired let me know; I can do that surgery.  (And did it when we made the same mistake...)
